### PR TITLE
✨ [MyDashboard/] Add stakeholder income

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-05-16 下午9 09 29
截圖 2023-05-16 下午9 09 20

### CODE REVIEW

**Comment by WeiJun0827** [comment]
@williamchong Do you think I should just integrate querying the user's income into `GET /user-stat`?

File: src/components/UserStats/IncomeDetailsDialog.vue [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      const items = [
+        {
+          text: this.$t('dashboard_button_type_sales'),
+          value: DETAILS_TYPE.SALES,
+          amount: Math.floor(this.totalSales * nanolike),
+        },
+        {
+          text: this.$t('dashboard_button_type_commissions'),
+          value: DETAILS_TYPE.COMMISSION,
+          amount: Math.floor(this.totalCommission * nanolike),
```
Comment (williamchong) [inline src/components/UserStats/IncomeDetailsDialog.vue:91]: Please use bignumber for nanolike => like conversions

File: src/components/UserStats/IncomeTable.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <div v-else class=" w-full my-[62px]">
+      <Label class="text-medium-gray" :text="$t('dashboard_table_no_data')" align="center" />
+    </div>
+  </div>
+</template>
+<script>
+import { formatNumberWithLIKE } from '~/util/ui';
+
+const DETAILS_TYPE = {
+  COMMISSION: 'commission',
```
Comment (williamchong) [inline src/components/UserStats/IncomeTable.vue:65]: Please call this `ROYALTY`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/pages/dashboard/index.vue [hunk 3]
```diff
@@ -300,13 +300,18 @@ export default {
...
-        ]);
+      try {
+        if (!this.getTotalSales && !this.getTotalRoyalty) {
+          await Promise.all([
+            this.walletFetchTotalSales(this.wallet),
+            this.walletFetchTotalRoyalty(this.wallet),
+          ]);
+        }
+      } catch (error) {
+        // no-op
```
Comment (williamchong) [inline src/pages/dashboard/index.vue:311]: Please `console.error` if error is not expected
or just don't have a `catch` block if the error does not need to be caught

**Comment by AuroraHuang22** [comment]
Is this better? @williamchong 
截圖 2023-06-05 下午3 38 56

**Comment by williamchong** [comment]
> Is this better? @williamchong 截圖 2023-06-05 下午3 38 56

Can we have the icon with an arrow?
https://mui.com/material-ui/material-icons/?query=new+window&selected=OpenInNew

**Comment by AuroraHuang22** [comment]
截圖 2023-06-05 下午4 09 01

File: src/components/UserStats/IncomeMenuItem.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          'rounded-[58px]',
+          'cursor-pointer',
+          'transition-all',
+          'duration-200',
+
+          'text-dark-gray bg-shade-gray hover:text-medium-gray',
+          { '!bg-like-green text-shade-gray shadow-sm hover:text-like-cyan-gray': item.isSelected },
+
+        ]"
+        @click="$emit('on-type-click',item.value)"
```
Comment (nwingt) [inline src/components/UserStats/IncomeMenuItem.vue:37]: Vue typically event emitter normally not start with `on` like React 
For example, `onClick` -> `click`

File: src/components/UserStats/IncomeDetailsDialog.vue [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    </div>
+  </Dialog>
+</template>
+
+<script>
+import { ellipsis } from '~/util/ui';
+import { convertToLIKEPrice } from '~/util/nft';
+import nftMixin from '~/mixins/nft';
+import { logTrackerEvent } from '~/util/EventLogger';
+
```
Comment (nwingt) [inline src/components/UserStats/IncomeDetailsDialog.vue:31]: Just sorting import that group by subfolder
```suggestion
import nftMixin from '~/mixins/nft';

import { logTrackerEvent } from '~/util/EventLogger';
import { convertToLIKEPrice } from '~/util/nft';
import { ellipsis } from '~/util/ui';
```

File: src/components/UserStats/MyDashboard.vue [hunk 6]
```diff
@@ -101,5 +119,20 @@ export default {
       default: '',
     },
   },
+  computed: {
+    itemStyle() {
```
Comment (nwingt) [inline src/components/UserStats/MyDashboard.vue:123]: should name `itemClasses`?

File: src/locales/en.json [hunk 7]
```diff
@@ -136,6 +136,14 @@
...
   "dashboard_portfolio_button": "Public Portfolio",
   "dashboard_title": "My Dashboard",
+  "dashboard_button_type_sales": "Sales Earnings",
+  "dashboard_button_type_royalties": "Royalties",
+  "dashboard_table_class_name": "NFT Name",
+  "dashboard_table_class_price": "Sales Revenue",
+  "dashboard_table_class_stake_percentage": "Profit-sharing Ratio",
+  "dashboard_table_class_net_earning": "Net Earnings",
+  "dashboard_table_no_data": "No Data",
+
```
Comment (nwingt) [inline src/locales/en.json:146]: Please sort it with JTM

File: src/locales/en.json [hunk 8]
```diff
@@ -402,8 +410,9 @@
   "nft_portfolio_page_state_collections": "Collections",
   "nft_portfolio_page_state_collectors": "Collectors",
   "nft_portfolio_page_state_creations": "Creations",
-  "nft_portfolio_page_state_sales": "Total Revenue (LIKE)",
-  "nft_portfolio_page_state_value": "Asset Value (LIKE)",
+  "nft_portfolio_page_state_sales": "Total Sales\n(LIKE)",
+  "nft_portfolio_page_state_value": "Asset Value\n(LIKE)",
+  "nft_portfolio_page_state_royalties": "Total Royalty\n(LIKE)",
```
Comment (nwingt) [inline src/locales/en.json:415]: Same here, please sort

File: src/store/modules/wallet.js [hunk 9]
```diff
@@ -199,6 +222,10 @@ const getters = {
       return count;
     }, 0);
   },
+  getTotalSales: state => state.totalSales,
+  getTotalRoyalty: state => state.totalRoyalty,
+  getSalesDetails: state => state.salesDetails,
+  getRoyaltyDetails: state => state.royaltyDetails,
```
Comment (nwingt) [inline src/store/modules/wallet.js:228]: Prefer name not starting with `get` as it is not a function
```suggestion
  walletTotalSales: state => state.totalSales,
  walletTotalRoyalty: state => state.totalRoyalty,
  walletSalesDetails: state => state.salesDetails,
  walletRoyaltyDetails: state => state.royaltyDetails,
```

**Review by nwingt — COMMENTED** [review]
Just code style comments, no big issue 🥳

**Comment by AuroraHuang22** [comment]
截圖 2023-06-06 上午9 10 12

**Review by nwingt — APPROVED** [review]

**Pull request merged**